### PR TITLE
Fix `Decimal('Infinity')` adaptation

### PR DESCRIPTION
**Pull request by edgarrmondragon**
Closes https://github.com/psycopg/psycopg2/issues/1724

### CODE REVIEW

File: tests/test_types_basic.py [hunk 1]
```diff
@@ -74,12 +74,15 @@ def testDecimal(self):
         self.failUnless(str(s) == "NaN", "wrong decimal quoting: " + str(s))
         self.failUnless(type(s) == decimal.Decimal,
                         "wrong decimal conversion: " + repr(s))
+
+    @testutils.skip_before_postgres(14, 0)
```
Comment (edgarrmondragon) [inline tests/test_types_basic.py:78]: `SELECT 'Infinity'::numeric AS foo` causes

```
psycopg2.errors.InvalidTextRepresentation: invalid input syntax for type numeric: "Infinity"
```

in Postgres 12 and 13.

Comment (edgarrmondragon) [reply tests/test_types_basic.py:78]: Oh, now that I think about it more we should probably rather check the server version when adapting `Decimal('Infinity')`, and fall back to `NaN'::numeric` on Postgres < 14.

Comment (dvarrazzo) [reply tests/test_types_basic.py:78]: Uhm... Thinking a bit harder abut it: is it even a good idea to change the way inf is converted? Yes, converting it to nan wasn't the right thing to do, but maybe there is code that assumes that behaviour.

I am reconsidering the opportunity of making this change.

Note: psycopg 3 doesn't check for version and doesn't convert inf -> nan. Trying to send an inf results in a server error, which is the right thing to do.

File: psycopg/adapter_pdecimal.c [hunk 2]
```diff
@@ -47,11 +47,24 @@ pdecimal_getquoted(pdecimalObject *self, PyObject *args)
...
+    check = PyObject_CallMethod(self->wrapped, "is_signed", NULL);
+    if (check == Py_True) {
+        res = Bytes_FromString("'-Infinity'::numeric");
+        goto end;
+    } else {
+        res = Bytes_FromString("'Infinity'::numeric");
+        goto end;
+    }
+
     /* is_finite() was introduced 2.5.1 < somewhere <= 2.5.4.
```
Comment (dvarrazzo) [inline psycopg/adapter_pdecimal.c:68]: Wow, I'm not sure this code is still used :smile:

**Review by dvarrazzo — CHANGES_REQUESTED** [review]

File: tests/test_types_basic.py (hunk 1 above)
Comment (edgarrmondragon) [reply tests/test_types_basic.py:78]: I agree. Either option (converting it or failing) is a potential breaking change. Happy to close this if you think we err on preserving current behavior.

**Comment by dvarrazzo** [comment]
Thank you very much for this MR, but as discussed it's probably not a good idea to do this in psycopg2.

**Pull request closed without merging**